src/components/Drawer.js

```diff
@@ -67,7 +67,12 @@ const Drawer = React.createClass({
     }
   },
 
-  _handleCloseClick () {
+  /**
+   * Animate the Drawer closed and then call the onClose callback.
+   *
+   * @returns {Promise} that is resolved when the animation finishes
+   */
+  close () {
```

Comment:
@tumentumurchudur using the `_` prefix is a common convention in javascript for private functions. I think it might be confusing for other javascript developers to call what looks like a private function so I would argue that it's better to keep it as `close`. Would you prefer having two functions a private `_handleCloseClick` which is used internally and called by the public `close` function?

Take a look at the `animate` function in this example from the React docs: https://facebook.github.io/react/tips/expose-component-functions.html